### PR TITLE
FLUT-7137 - [Feature] UG documentation has been added to DataGrid for the filter icon style and popup menu customizing features.

### PR DESCRIPTION
**Pull request by Tamilarasan-Paranthaman**
## Description ##

* UG documentation has been added to DataGrid for the `filter icon style` and `popup menu customizing` features.

### CODE REVIEW

**Review by NeelakandanKannan — CHANGES_REQUESTED** [review]
Get back to me with the changes.

File: Flutter/datagrid/filtering.md [hunk 1]
```diff
@@ -713,4 +713,426 @@ Widget build(BuildContext context) {
...
 {% endhighlight %}
-{% endtabs %}
\ No newline at end of file
+{% endtabs %}
+
+## Customizing the Filter Popup Menu Options
+
+The `SfDataGrid` provides the support to customizing the following options in the filter popup menu.
+
+### Filter Mode
```
[the remark this answers is not in the document]
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/filtering.md:722]: Modified

File: Flutter/datagrid/filtering.md [hunk 2]
```diff
@@ -713,4 +713,426 @@ Widget build(BuildContext context) {
 }
 
 {% endhighlight %}
-{% endtabs %}
\ No newline at end of file
+{% endtabs %}
+
+## Customizing the Filter Popup Menu Options
```
[the remark this answers is not in the document]
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/filtering.md:718]: Modified

File: Flutter/datagrid/filtering.md [hunk 3]
```diff
@@ -713,4 +713,426 @@ Widget build(BuildContext context) {
...
\ No newline at end of file
+{% endtabs %}
+
+## Customizing the Filter Popup Menu Options
+
+The `SfDataGrid` provides the support to customizing the following options in the filter popup menu.
+
+### Filter Mode
+
+The Filter popup mode for the column can be set by using the `FilterPopupMenuOptions.filterMode` in `GridColumn.filterPopupMenuOptions` property.
```
[the remark this answers is not in the document]
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/filtering.md:724]: Modified

File: Flutter/datagrid/filtering.md [hunk 4]
```diff
@@ -713,4 +713,426 @@ Widget build(BuildContext context) {
...
+    ),
+  );
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="flutter datagrid hiding clear filter options in popup menu" src="images/filtering/flutter-datagrid-hide-clear-filter-option.png" width="400"/>
+
+### Hiding column name
```
[the remark this answers is not in the document]
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/filtering.md:885]: I have changed

**Review by NeelakandanKannan — CHANGES_REQUESTED** [review]
Get back to me with the changes.

**Review by NeelakandanKannan — APPROVED** [review]
The changes are fine.

**Pull request merged**